pydantic/_internal/_validators.py

```diff
@@ -346,35 +346,38 @@ def _extract_decimal_digits_info(decimal: Decimal) -> tuple[int, int]:
     Though this could be divided into two separate functions, the logic is easier to follow if we couple the computation
     of the number of decimals and digits together.
     """
-    decimal_tuple = decimal.as_tuple()
-    if not isinstance(decimal_tuple.exponent, int):
-        raise TypeError(f'Unable to extract decimal digits info from supplied value {decimal}')
-    exponent = decimal_tuple.exponent
-    num_digits = len(decimal_tuple.digits)
-
-    if exponent >= 0:
-        # A positive exponent adds that many trailing zeros
-        # Ex: digit_tuple=(1, 2, 3), exponent=2 -> 12300 -> 0 decimal places, 5 digits
-        num_digits += exponent
-        decimal_places = 0
-    else:
-        # If the absolute value of the negative exponent is larger than the
-        # number of digits, then it's the same as the number of digits,
-        # because it'll consume all the digits in digit_tuple and then
-        # add abs(exponent) - len(digit_tuple) leading zeros after the decimal point.
-        # Ex: digit_tuple=(1, 2, 3), exponent=-2 -> 1.23 -> 2 decimal places, 3 digits
-        # Ex: digit_tuple=(1, 2, 3), exponent=-4 -> 0.0123 -> 4 decimal places, 4 digits
-        decimal_places = abs(exponent)
-        num_digits = max(num_digits, decimal_places)
+    try:
+        decimal_tuple = decimal.as_tuple()
 
-    return decimal_places, num_digits
```

Comment:
Can we leave this as is? What `AttributeError` might occur here?

Reply:
Since

Reply:
If the `decimal` argument is not a `Decimal` type, which is accepted as Any in the root function, an `AttributeError` might occur because the as_tuple method may not exist

Reply:
Would it be better to change the argument type to `Any` and add another `assert` to check if the type is `Decimal`? 🤔

Reply:
Makes sense, thanks for the improvement!